src/cljs/commiteth/common.cljs

```diff
@@ -12,16 +12,24 @@
                    :on-change #(reset! val-ratom (-> % .-target .-value))})]))
 
 (defn dropdown [props title val-ratom items]
-  (fn []
-    (if (= 1 (count items))
-      (reset! val-ratom (first items)))
-    [:select.ui.basic.selection.dropdown
-     (merge props {:on-change
-                   #(reset! val-ratom (-> % .-target .-value))})
-     (doall (for [item items]
-              ^{:key item} [:option
-                            {:value item}
-                            item]))]))
+  "If val-ratom is set, preselect it in the dropdown.
+   Add value of val-ratom if it's missing from items list.
+   Otherwise, prepend title as a disabled option"
+  (let [items (cond-> items
+                  (and @val-ratom
+                       (not (contains? (set items) @val-ratom)))
+                  (conj @val-ratom)
+                  (not @val-ratom)
+                  (conj title))]
```

Comment:
Are we requiring the item coll to be a list? Since `conj` would behave differently with other types of colls. How about putting it the first element somehow even if it is not the most efficient way, so that at least we comply with semantics if other colls are used?

Reply:
Agree. Using `into` instead.